File: binary__linear/binary.js
function binary(arr,element,size){
    var low = 0;
    var high = size - 1;
    while(low<=high){
        var mid = Math.floor((high+low)/2)
        if(arr[mid]== element){
            return mid
        }
        if (arr[mid] < element){
            low = mid +1 
        }
        else{
            high = mid - 1
        }
    }
    return -1
}
array = [1,2,3,4,5,6,7,8,9,10]
size = array.length
element = 7
console.log(binary(array,element,size)) 

File: binary__linear/binary.py
import math
def binary(list,element,size0):
    low = 0
    high = size0 - 1
    while low<=high:
        mid = math.floor((high- low) / 2) 
        if list[mid] == element:
            return mid
        if list[mid]< element:
            low = mid +1
        else:
            high = mid-1
    return -1

arr = [1,2,3,4,5,6,7,8,9,10]
element = 5
size0 = len(arr)
result = binary(arr,element,size0)
if result != -1:
    print("Element is present at index " + str(result))